Solutions/Corvus.Testing.AzureFunctions/Corvus/Testing/AzureFunctions/FunctionsController.cs

```diff
@@ -207,8 +227,32 @@ private static void KillProcessAndChildren(int pid)
 
             try
             {
-                var proc = Process.GetProcessById(pid);
-                proc.Kill();
+                bool failedDueToAccessDenied;
+                int accessDenyRetryCount = 0;
+                const int MaxAccessDeniedRetries = 3;
+
+                do
+                {
+                    failedDueToAccessDenied = false;
+
+                    var proc = Process.GetProcessById(pid);
```

Comment:
Should we not check that the process has not, in fact, successfully died already here - i.e. no longer exists, and exit happily)?

Reply:
We did already have code that handled this:

```cs
catch (ArgumentException)
{
    // Process already exited.
}
```

However, the control flow was rather contorted - the relevant `catch` block was outside of the retry loop.

I've now moved it right in so it's clear we're only handling the `ArgumentException` from that one API call. (This also rules out accidentally handling it from places we failed to realised it would emerge from.)